substrate/frame/utility/src/lib.rs

```diff
@@ -454,6 +465,71 @@ pub mod pallet {
 			let res = call.dispatch_bypass_filter(frame_system::RawOrigin::Root.into());
 			res.map(|_| ()).map_err(|e| e.error)
 		}
+
+		#[pallet::call_index(6)]
+		#[pallet::weight({
+			let main_dispatch_info = main.get_dispatch_info();
+			let fallback_dispatch_info = fallback.get_dispatch_info();
+			(
+				T::WeightInfo::if_else()
+					.saturating_add(main_dispatch_info.call_weight)
+					.saturating_add(fallback_dispatch_info.call_weight),
+				main_dispatch_info.class,
+			)
+		})]
+		pub fn if_else(
+			origin: OriginFor<T>,
+			main: Box<<T as Config>::RuntimeCall>,
+			fallback: Box<<T as Config>::RuntimeCall>,
+		) -> DispatchResultWithPostInfo {
+			// Do not allow the `None` origin.
+			if ensure_none(origin.clone()).is_ok() {
+				return Err(BadOrigin.into());
+			}
+
+			let is_root = ensure_root(origin.clone()).is_ok();
+
+			// Track the weights
+			let mut weight = Weight::zero();
```

Comment:
This should be initialized with the `if_else` base weight:
```suggestion
			let mut weight = T::WeightInfo::if_else();
```

Reply:
could you explain this a bit more?

Reply:
You could argue that the weight of `if_else` call is just main or main + fallback, and actually I would agree because apart from those we only do negligible operations like math operation and deposit event.

But we do have the benchmark: `if_else` which provides the weight for this operation so why not use it and be safe in case the code changes a bit.

In the pre dispatch weight we do: `if_else + main + fallback`.
But in the post dispatch weight do `main` |  `main + fallback` and ignore the `if_else` base cost.

I think for more safety and consistency we should do post dispatch being: `if_else + main` | `if_else + main + fallback`.